### PR TITLE
exit with status 1 if --help is called on an invalid command

### LINKED ISSUE #28329: Non-existent subcommand exit code behavior changed in 1.13+ (1.13.0-rc1)
On previous versions of Docker, the validity of a subcommand could be determined by invoking `docker <subcommand> --help` and checking the exit code.

For existing commands (such as `docker image --help`), the exit code was zero indicating success.

For invalid commands (such as `docker invalid --help`), the exit code was non-zero, as expected.

In v1.13.0-rc1, this is no longer the case:
```console
$ ./docker-1.12.3 images --help > /dev/null; echo $?
0
$ ./docker-1.12.3 invalid --help > /dev/null; echo $?
docker: 'invalid' is not a docker command.
See 'docker --help'.
1
$ ./docker-1.13.0-rc1 images --help > /dev/null; echo $?
0
$ ./docker-1.13.0-rc1 invalid --help > /dev/null; echo $?
0
```

Is this behavior change expected or accidental?

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #28329

Needs vendor

### CODE REVIEW

File: cmd/docker/docker_test.go [hunk 1]
```diff
@@ -16,14 +17,16 @@ func TestClientDebugEnabled(t *testing.T) {
...
-	}
-	if logrus.GetLevel() != logrus.DebugLevel {
-		t.Fatalf("expected logrus debug level, got %v", logrus.GetLevel())
-	}
+func TestExitStatusForInvalidSubcommandWithHelpFlag(t *testing.T) {
+	discard := ioutil.Discard
+	cmd := newDockerCommand(command.NewDockerCli(os.Stdin, discard, discard))
+	cmd.SetArgs([]string{"invalid", "--help"})
+	err := cmd.Execute()
+	assert.Error(t, err, "'invalid' is not a docker command.")
```
Comment (tianon) [inline cmd/docker/docker_test.go]: Do we already have tests covering the inverse? (ie, `--help` on a valid commands returning 0)

**Review by tianon — COMMENTED** [review]
:heart:!

File: cmd/docker/docker_test.go (hunk 1 above)
Comment (dnephin) [inline cmd/docker/docker_test.go]: Yes, I we have extensive tests on `--help` in `integration-cli/docker_cli_help_test.go`

Comment (vdemeester) [inline cmd/docker/docker_test.go]: @dnephin tests that I intend to move to unit tests real soon :angel:

**Comment by runcom** [comment]
Just tested, LGTM
```sh
$ docker images --help > /dev/null; echo $?                                                                                       
0
$ docker invalid --help > /dev/null; echo $?
docker: 'invalid' is not a docker command.
See 'docker --help'.
1
```

**Comment by dnephin** [comment]
hmm, so this isn't going to work either. It fails when you do `docker service --help`, because search expects an argument.

**Pull request closed without merging**